src/NLog/LayoutRenderers/MdlcLayoutRenderer.cs

```diff
@@ -58,8 +59,14 @@ public class MdlcLayoutRenderer : LayoutRenderer
         /// <param name="logEvent">Logging event.</param>
         protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         {
-            var message = MappedDiagnosticsLogicalContext.Get(Item);
-            builder.Append(message);
+            //don't use MappedDiagnosticsLogicalContext.Get to ensure we are not locking the Factory (indirect by LogManager.Configuration).
+            var o = MappedDiagnosticsLogicalContext.GetObject(this.Item);
+            var formatProvider = logEvent.FormatProvider;
+            if (formatProvider == null && LoggingConfiguration != null)
+            {
+                formatProvider = LoggingConfiguration.DefaultCultureInfo;
+            }
+            builder.Append(Convert.ToString(o, formatProvider));
```

Comment:
Wouldn't it benefit if we add this logic to the base class and have it is a property?